### PR TITLE
Async Upgrades

### PR DESCRIPTION
**Pull request by elijahbenizzy**
- [x] #1023
- [x] #1022 
- [x] Adds option to construct async driver synchronously with the builder

The test tests at the execute and not raw_execute() level, as that is more indicative of the contract.

## Checklist

- [x] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [x] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future TODOs are captured in comments
- [x] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
Thanks to [@rwr](https://github.com/DAGWorks-Inc/hamilton/issues?q=is%3Aissue+is%3Aopen+author%3Arwhitten577)
for the bug finds! 

This has been released (and verified) in `1.71.0rc1` -- we will be releasing as part of the next cut.

File: ui/frontend/src/components/dashboard/Runs/Runs.tsx [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { Loading } from "../../common/Loading";
 import { RunSummary } from "./RunSummary";
 import { skipToken } from "@reduxjs/toolkit/dist/query";
 
-export const MAX_RUNS_QUERIED = 1000;
```
Comment (skrawcz) [inline ui/frontend/src/components/dashboard/Runs/Runs.tsx:11]: this meant to be here? maybe an env variable to override?

File: ui/admin.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def _build_ui():
     cmd = "npm run build --prefix ui/frontend"
     _command(cmd, capture_output=False)
     # wiping the old build if it exists
-    cmd = "rm -rf ui/backend/build"
+    cmd = "rm -rf ui/backend/server/build"
```
Comment (skrawcz) [inline ui/admin.py:70]: this right?

**Review by skrawcz — APPROVED** [review]
just need minor changes...

**Pull request merged**